src/__tests__/resources/asset.integration.spec.ts

```diff
@@ -0,0 +1,219 @@
+import { sleepPromise } from '../../utils';
+import { randomInt, setupLoggedInClient } from '../testUtils';
+
+describe('Asset', () => {
+  let client: CogniteClient;
+  beforeAll(() => {
+    client = setupLoggedInClient();
+  });
+  const rootAsset = {
+    name: 'test-root',
+    description: 'Root asset for cognitesdk-js test',
+    metadata: {
+      refId: 'test-root',
+    },
+    externalId: 'test-root' + randomInt(),
+  };
+
+  const childAsset = {
+    // tslint:disable-next-line:no-duplicate-string
+    name: 'test-child',
+    description: 'Child asset for cognitesdk-js test',
+    metadata: {
+      refId: 'test-child',
+    },
+    parentExternalId: rootAsset.externalId,
+    externalId: 'test-child' + randomInt(),
+  };
+  let assets: AssetList;
+
+  test('create', async () => {
+    assets = await client.assets.create([childAsset, rootAsset]);
+    expect(assets[0].createdTime).toBeInstanceOf(Date);
+    expect(assets[0].lastUpdatedTime).toBeInstanceOf(Date);
+  });
+
+  test('update', async () => {
+    await client.assets.update([
+      {
+        id: assets[0].id,
+        update: {
+          name: { set: 'New name' },
+        },
+      },
+    ]);
+  });
+
+  test('delete', async () => {
+    await assets[0].delete();
+  });
+
+  test('list', async () => {
+    await client.assets
+      .list({
+        filter: {
+          name: rootAsset.name,
+          createdTime: { min: 0, max: Date.now() },
+        },
+      })
+      .autoPagingToArray({ limit: 100 });
+  });
+
+  test('search for root test asset', async () => {
+    const result = await client.assets.search({
+      search: {
+        name: rootAsset.name,
+      },
+    });
+    const asset = result[0];
+    expect(rootAsset.name).toBe(asset.name);
+    expect(rootAsset.description).toBe(asset.description);
+  });
+
+  describe('Asset Class', () => {
+    const newRootAsset = {
+      name: 'test-root',
+    };
+    const newChildAsset = {
+      name: 'test-child',
+    };
+    const newGrandChildAsset = {
+      name: 'test-grandchild',
+    };
+
+    test('create 1001 assets sequencially', async () => {
+      const childArr = new Array(1000).fill(newChildAsset);
+      const createdAssets = await client.assets.create([
+        newRootAsset,
+        ...childArr,
+      ]);
+      expect(createdAssets.length).toBe(1001);
+      await client.assets.delete(
+        createdAssets.slice(1).map(asset => ({ id: asset.id }))
+      );
+      await client.assets.delete([{ id: createdAssets[0].id }]);
+    });
+
+    test('parent', async () => {
+      const newRoot = {
+        ...newRootAsset,
+        externalId: 'test-root' + randomInt(),
+      };
+      const newChild = {
+        ...newChildAsset,
+        parentExternalId: newRoot.externalId,
+      };
+      const createdAssets = await client.assets.create([newRoot, newChild]);
+      const parent = await createdAssets[1].parent();
+      if (parent) {
```

Comment:
Yes, thing is I can't do `parent.id` because `parent` can be `Asset | null` at that point, that's why I had to put `if` statement there. Could just remove the whole id-testing and do instanceOf.